tests/test_frameDict.py

```diff
@@ -0,0 +1,299 @@
+        # Test using arbitrary values for x,y,z
+        x = 75.1
+        y = -53.2
+        z = 0.123
+        frame1 = ast.Frame(3)
+        permMap = ast.PermMap([1, 2, -1], [1, 2], [z])
+        frame2 = ast.Frame(2)
+        frameDict = ast.FrameDict(frame1, permMap, frame2)
+        self.assertAlmostEqual(frameDict.applyForward([x, y, z]), [x, y])
+        self.assertAlmostEqual(frameDict.applyInverse([x, y]), [x, y, z])
+
+        # permuting the axes of the current frame also permutes the mapping
+        frameDict.permAxes([2, 1])
+        self.assertAlmostEqual(frameDict.applyForward([x, y, z]), [y, x])
+        self.assertAlmostEqual(frameDict.applyInverse([x, y]), [y, x, z])
+
+        # permuting again puts things back
+        frameDict.permAxes([2, 1])
+        self.assertAlmostEqual(frameDict.applyForward([x, y, z]), [x, y])
+        self.assertAlmostEqual(frameDict.applyInverse([x, y]), [x, y, z])
+
+    def testFrameDictSetBaseCurrent(self):
+        frameDict = ast.FrameDict(self.frame1, self.zoomMap, self.frame2)
+        self.assertEqual(frameDict.base, 1)
+        self.assertEqual(frameDict.current, 2)
+        self.assertEqual(frameDict.getIndex("frame1"), 1)
+        self.assertEqual(frameDict.getIndex("frame2"), 2)
+
+        indata = np.array([
+            [0.0, 0.1, -1.5],
+            [5.1, 0.0, 3.1],
+        ])
+        predictedOut1 = indata.copy() * self.zoom
+        assert_allclose(frameDict.applyForward(indata), predictedOut1)
+
+        frameDict.setCurrent("FRAME1")
+        self.assertEqual(frameDict.base, 1)
+        self.assertEqual(frameDict.current, 1)
+        self.assertEqual(frameDict.getIndex("FRAME1"), 1)
+        self.assertEqual(frameDict.getIndex("FRAME2"), 2)
+
+        predictedOutput2 = indata.copy()
+        assert_allclose(frameDict.applyForward(indata), predictedOutput2)
+
+        frameDict.setBase("FRAME2")
+        self.assertEqual(frameDict.base, 2)
+        self.assertEqual(frameDict.current, 1)
+        self.assertEqual(frameDict.getIndex("FRAME1"), 1)
+        self.assertEqual(frameDict.getIndex("FRAME2"), 2)
+
+        predictedOutput3 = indata.copy() / self.zoom
+        assert_allclose(frameDict.applyForward(indata), predictedOutput3)
+
+    def testFrameDictSetDomain(self):
+        frameDict = ast.FrameDict(self.frame1, self.zoomMap, self.frame2)
+        frameDict.setCurrent("FRAME1")
+        frameDict.setDomain("NEWFRAME1")
+        self.assertEqual(set(frameDict.getAllDomains()), set(["NEWFRAME1", "FRAME2"]))
+        self.assertEqual(frameDict.getIndex("newFrame1"), 1)
+        self.assertEqual(frameDict.getIndex("FRAME2"), 2)
+
+        frameDict.setCurrent("FRAME2")
+        frameDict.setDomain("NEWFRAME2")
+        self.assertEqual(set(frameDict.getAllDomains()), set(["NEWFRAME1", "NEWFRAME2"]))
+        self.assertEqual(frameDict.getIndex("NEWFRAME1"), 1)
+        self.assertEqual(frameDict.getIndex("NEWFRAME2"), 2)
+
+
```

Comment:
Can you add tests for `getFrame`, `getMapping`, etc. that try to use a missing domain?